File: Editorial/0000-0100/0073_Set_Matrix_Zeroes/C/1_Additional_Memory_Approach.c
void setZeroes(int** matrix, int matrixSize, int* matrixColSize) {
    int R = matrixSize;
    int C = *matrixColSize;
    int rows[R], cols[C];
    memset(rows, 0, sizeof(rows));
    memset(cols, 0, sizeof(cols));
    // Essentially, we mark the rows and columns that are to be made zero
    for (int i = 0; i < R; i++) {
        for (int j = 0; j < C; j++) {
            if (matrix[i][j] == 0) {
                rows[i] = 1;
                cols[j] = 1;
            }
        }
    }
    // Iterate over the array once again and using the rows and cols sets,
    // update the elements
    for (int i = 0; i < R; i++) {
        for (int j = 0; j < C; j++) {
            if (rows[i] == 1 || cols[j] == 1) {
                matrix[i][j] = 0;
            }
        }
    }
}

File: Editorial/0000-0100/0073_Set_Matrix_Zeroes/JavaScript/1_Additional_Memory_Approach.js
var setZeroes = function (matrix) {
    const rows = new Set();
    const cols = new Set();
    // Essentially, we mark the rows and columns that are to be made zero
    for (let i = 0; i < matrix.length; i++) {
        for (let j = 0; j < matrix[0].length; j++) {
            if (matrix[i][j] == 0) {
                rows.add(i);
                cols.add(j);
            }
        }
    }
    // Iterate over the array once again and using the rows and cols sets, update the elements
    for (let i = 0; i < matrix.length; i++) {
        for (let j = 0; j < matrix[0].length; j++) {
            if (rows.has(i) || cols.has(j)) {
                matrix[i][j] = 0;
            }
        }
    }
};
